src/physics/utilities/equation_solver.cpp

```diff
@@ -58,11 +58,38 @@ std::unique_ptr<mfem::IterativeSolver> EquationSolver::buildIterativeLinearSolve
 }
 
 std::unique_ptr<mfem::NewtonSolver> EquationSolver::buildNewtonSolver(MPI_Comm                         comm,
-                                                                      const NonlinearSolverParameters& nonlin_params,
-                                                                      mfem::Solver&                    lin_solver)
+                                                                      const NonlinearSolverParameters& nonlin_params)
 {
-  auto newton_solver = std::make_unique<mfem::NewtonSolver>(comm);
-  newton_solver->SetSolver(lin_solver);
+  std::unique_ptr<mfem::NewtonSolver> newton_solver;
+
+  if (nonlin_params.nonlin_solver == NonlinearSolver::BasicNewton) {
+    newton_solver = std::make_unique<mfem::NewtonSolver>(comm);
+  }
+  // KINSOL
+  else {
+#ifdef MFEM_USE_SUNDIALS
```

Comment:
Do we need the ifdefs here or do we assume that Sundials is always available?

Reply:
This PR adds sundials as an optional dependency (though I have no problem with making it a requirement)

Reply:
Hmmm. We should probably keep it as optional because simpler linear quasi-static physics (e.g. LiDO in a bunch of cases) won't need it.